extra/src/main/java/aima/extra/search/pqueue/uninformed/GraphPriorityQueueSearch.java

```diff
@@ -45,19 +46,19 @@
 		// node <- a node with STATE = problem.INITIAL-STATE
 		Node<A, S> node = newRootNode(problem.initialState());
 		// frontier <- a priority queue, with node as the only element
-		Queue<Node<A, S>> frontier = newFrontier(node);
-		// explored <- an empty set
+        PriorityQueue<Node<A, S>> pqFrontier = newFrontierPriority(node, getNodeComparator());
+        // explored <- an empty set

```

Comment:
By explicitly using PriorityQueue<> you have now make the code dependent on a class not an interface, which would make it more difficult to switch out different priority queue implementations. That is why 'Queue' is used in the existing code.